ecosystem/indexer-grpc/indexer-grpc-data-service/src/lib.rs

```diff
@@ -1,6 +1,21 @@
 // Copyright © Aptos Foundation
 // SPDX-License-Identifier: Apache-2.0
 
-#![forbid(unsafe_code)]
+use aptos_protos::transaction::testing1::v1::Transaction;
+use prost::Message;
 
+pub mod metrics;
 pub mod service;
+
+pub fn calculate_data_latency_from_encoded_pb(encoded_pb: &str) -> Option<u64> {
+    let transaction_pb = base64::decode(encoded_pb).ok()?;
+    let transaction = Transaction::decode(transaction_pb.as_slice()).ok()?;
+    let current_timestamp = std::time::SystemTime::now()
+        .duration_since(std::time::UNIX_EPOCH)
+        .ok()?;
+    let current_timestamp = current_timestamp.as_nanos() as u64;
+    transaction.timestamp.map(|timestamp| {
+        (current_timestamp - timestamp.nanos as u64 - timestamp.seconds as u64 * 1_000_000_000)
+            / 1_000_000
+    })
+}
```

Comment:
move to utils folder and split function.

Reply:
done